dashboard/src/components/DeploymentForm/DeploymentForm.tsx

```diff
@@ -267,6 +267,7 @@ export default function DeploymentForm() {
                 setValues={handleValuesChange}
                 appValues={appValues}
                 setValuesModified={setValuesModifiedTrue}
+                formRef={formRef}
```

Comment:
It seems that `formRef` is only used for performing a `requestSubmit()` in the parent.
Why not passing a plain callback function as parameter instead of this reference?

Reply:
It was introduced in one of my previous attempts when working on this. The idea is to trigger a plain normal HTML form "submit" event from a child, this way we leverage the built-in HTML validations (like `required`, `max`, `min`, `pattern`).

Anyway, just to double-check, I'm trying locally to replace the `ref` with the `handleDeploy()` function directly to ensure the `ref` is really needed or not.

Reply:
Exactly, that's why I left the hacky `ref`, otherwise the HTML are not taken into account :(

![nohtmlval](https://user-images.githubusercontent.com/11535726/194037421-532ccd80-29b0-4119-9f65-58b78c4a4161.gif)